pip/compat.py

```diff
@@ -41,7 +46,14 @@ def console_to_str(s):
         try:
             return s.decode(sys.__stdout__.encoding)
         except UnicodeDecodeError:
-            return s.decode('utf_8')
+            if WINDOWS:
+                try:
+                    from ctypes import cdll
+                    return s.decode("cp" + str(cdll.kernel32.GetACP()))
```

Comment:
Should we be using the ANSI codepage here? As we're typically running a CLI subprocess, should it not be the OEM codepage that it will use? It's also worth noting that under Python 3.6, ```sys.__stdout__.encoding``` will be UTF-8 (as Python 3.6 changed to use Unicode output on the console) but this will not necessarily match what a process run via ```subprocess.call``` would see.

This whole area is something of a gross hack anyway, as there's really no way of knowing what encoding a subprocess might have chosen to use for its output. The best we can do is guess.

Maybe what would be better would be:

1. Try ```sys.__stdout__.encoding```.
2. Try UTF-8 (just because we already do that, I don't really know the justification, maybe it's for Unix)
3. Use ASCII, with a suitable error handler (backslashreplace for Python 3.5+, replace for earlier versions)

Use the first result that doesn't give ```UnicodeDecodeError``` (option 3 never should).

As far as I can see, the only way we use this function is to display the output of the ```setup.py``` invocation, so a little bit of information loss or unreadability (the replace/backslashreplace option) is OK - certainly better than a decoding error.